src/hooks/usePools.ts

```diff
@@ -105,8 +105,12 @@ export function usePools(
       if (!tokens) return [PoolState.INVALID, null]
       const [token0, token1, fee] = tokens
 
+      if (!slot0s[index]) return [PoolState.INVALID, null]
```

Comment:
nit: do the check before the destructure to make it apparent why you're doing them:

```
if (!slots0[index]) ...
const {...} = slots0[index]

if (!liquidities[index]) ...
...
```